crates/solvers/src/domain/solver/dex.rs

```diff
@@ -0,0 +1,78 @@
+//! A simple solver that matches orders directly with swaps from the external
+//! DEX and DEX aggregator APIs.
+
+use {
+    crate::{
+        domain::{
+            auction,
+            dex::{self, slippage},
+            order,
+            solution,
+        },
+        infra,
+    },
+    tracing::Instrument,
+};
+
+pub struct Dex {
+    /// The DEX API client.
+    pub dex: infra::dex::Dex,
+
+    /// The slippage configuration to use for the solver.
+    pub slippage: slippage::Limits,
+}
+
+impl Dex {
+    pub async fn solve(&self, auction: auction::Auction) -> Vec<solution::Solution> {
+        // TODO: order prioritization, skip liquidity orders, concurrency.
+        let prices = slippage::Prices::for_auction(&auction);
+
+        let mut solutions = Vec::new();
+        for order in auction.orders {
+            let span = tracing::info_span!("solve", order = %order.uid);
+            if let Some(solution) = self
+                .solve_order(order, &prices, auction.gas_price)
+                .instrument(span)
+                .await
+            {
+                solutions.push(solution);
+            }
+        }
+
+        solutions
+    }
+
+    async fn solve_order(
+        &self,
+        order: order::Order,
+        prices: &slippage::Prices,
+        gas: auction::GasPrice,
+    ) -> Option<solution::Solution> {
+        let query = dex::Order::new(&order);
+        let slippage = self.slippage.relative(&query.amount(), prices);
+
+        let swap = match self.dex.swap(&query, &slippage, gas).await {
+            Ok(swap) => swap,
+            Err(infra::dex::Error::NotFound) => {
+                tracing::debug!("cound not find swap for order");
+                return None;
+            }
```

Comment:
How about just `tracing::debug(err, "skipping order")`? The error already knows what kind of error it is. We don't need to match on all avariants to log a custom a message.